### PR TITLE
Creating module for database utilities.

### PR DESCRIPTION
**Pull request by drugilsberg**
Implementing an interface for pymongo.

This is a draft only because I would like to make sure that it's useful to have the module here.
This is why I did not configure any of the variables for integration testing and CI will fail.
Will configure it in case the PR moves after the draft stage.

Signed-off-by: Matteo Manica <drugilsberg@gmail.com>

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -23,6 +23,8 @@ install_requires =
     click>=7.0
     diskcache>=5.2.1
     typing-extensions>=4.1.1
+    pymongo>=3.9.0
+    pydantic[dotenv]>=1.9.0
```
Comment (avaucher) [inline setup.cfg:27]: Minor comment: I usually try to keep the dependencies in alphabetical order, to make maintenance easier, if you can just change that.

Comment (drugilsberg) [reply setup.cfg:27]: sure will do

File: tests/test_databases_pymongo.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import os
```
[the remark this answers is not in the document]
Comment (drugilsberg) [reply tests/test_databases_pymongo.py:1]: Thanks, let me start replying herE:

- sure will add the comment (basically they patch the env to simulate potential missing variables or setting values for the certificate)
- here I was a bit conflicted, basically in the CI I would set some secrets for connecting to an actual cluster and the if/else logic would be by-passed. I decided to have it there to make sure that a person developing locally rxn-utilities does not need to have the env variables for pymongo set to pass the unit tests. That said I think I will parameterize the tests to simply skip the ones that require the env variable defined.

File: tests/test_databases_pymongo.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    "RXN_MONGO_URI" not in os.environ, reason="environment_not_configured_for_test"
+)
+def test_get_client_with_tls_ca_cert(mock_mongo_certificate_env):
+    _instantiate_client_and_check_server_info()
+
+
+def test_get_pymongo_settings_with_no_mongo_uri(mock_no_mongo_uri_env):
+    pymongo_settings = PyMongoSettings()
+    assert isinstance(pymongo_settings, PyMongoSettings)
+    assert pymongo_settings.mongo_uri == os.environ.get("RXN_MONGO_URI", None)
```
Comment (avaucher) [inline tests/test_databases_pymongo.py:65]: Could this be directly
```suggestion
    assert pymongo_settings.mongo_uri is None
```
?

File: tests/test_databases_pymongo.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (avaucher) [reply tests/test_databases_pymongo.py:1]: Ok, I think I start to understand.

The last thing that is unclear is `test_get_client_with_tls_ca_cert` vs `test_get_client`, is it just that one tests the connection when a certificate is available and the other one if there is no certificate?

File: tests/test_databases_pymongo.py (hunk 3 above)
Comment (drugilsberg) [reply tests/test_databases_pymongo.py:65]: sure will update, it was a leftover :)

Comment (drugilsberg) [reply tests/test_databases_pymongo.py:65]: regarding the comment below, correct that is the main difference.

**Review by avaucher — APPROVED** [review]

**Pull request merged**